recipes/cassandra-cpp-driver/all/patches/2.15.3/fix-cmake.patch

```diff
@@ -0,0 +1,466 @@
+ CMakeLists.txt                               |  55 ++-----
+ src/CMakeLists.txt                           | 211 +++++++++++----------------
+ src/third_party/curl/CMakeLists.txt          |  16 +-
+ src/third_party/hdr_histogram/CMakeLists.txt |  15 +-
+ src/third_party/http-parser/CMakeLists.txt   |  15 +-
+ src/third_party/minizip/CMakeLists.txt       |  19 +--
+ 6 files changed, 114 insertions(+), 217 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index c6150948b..b2bee1aa7 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,5 +1,8 @@
+ cmake_minimum_required(VERSION 2.8.12)
+-project(cassandra C CXX)
++# project(cassandra C CXX)
++
++# include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
++# conan_basic_setup()
+ 
+ set(CASS_ROOT_DIR ${CMAKE_CURRENT_SOURCE_DIR})
+ set(CASS_SRC_DIR "${CASS_ROOT_DIR}/src")
+@@ -27,61 +30,23 @@ endif()
+ # Options
+ #---------------
+ 
+-option(CASS_BUILD_EXAMPLES "Build examples" OFF)
+-option(CASS_BUILD_INTEGRATION_TESTS "Build integration tests" OFF)
+-option(CASS_BUILD_SHARED "Build shared library" ON)
+-option(CASS_BUILD_STATIC "Build static library" OFF)
+-option(CASS_BUILD_TESTS "Build tests" OFF)
+-option(CASS_BUILD_UNIT_TESTS "Build unit tests" OFF)
+-option(CASS_DEBUG_CUSTOM_ALLOC "Debug custom allocator" OFF)
```

Comment:
Instead of removing these options and carrying a heavy patch,
I think you can set the definitions in the `conanfile`.
e.g.
```python
cmake.definitions["CASS_BUILD_EXAMPLES"] = False
```

Reply:
Setting it this way will make the patch much lighter, I think.
Because you don't have to re-do the add_library for shared/static thing.

Reply:
[Fixed at](https://github.com/conan-io/conan-center-index/pull/3110/commits/f6165087f3123b0c08c88e4f1040f3789dd9a607)

Please notice: I did not revert the part where I can choose wich lib to build shared/static.